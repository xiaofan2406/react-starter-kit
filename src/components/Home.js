import React from 'react';
import { css } from '@emotion/core';
import { spacing } from 'styles';
import { Anchor } from 'widgets';

const Home = () => (
  <div>
    <div
      css={css`
        font-size: 48px;
        text-align: center;
        margin-bottom: ${spacing.breath}px;
      `}
    >
      Hello world!
    </div>
    <table
      css={css`
        margin-top: ${spacing.breath}px;
        & td {
          padding: ${spacing.internal}px;
        }
      `}
    >
      <tbody>
        <tr>
          <td>React</td>
          <td>
            <Anchor href="https://reactjs.org/docs/thinking-in-react.html">
              Docs
            </Anchor>
          </td>
          <td>
            <Anchor href="https://github.com/facebook/react/">Github</Anchor>
          </td>
        </tr>
        <tr>
          <td>Reach Router</td>
          <td>
            <Anchor href="https://reach.tech/router">Docs</Anchor>
          </td>
          <td>
            <Anchor href="https://github.com/reach/router">Github</Anchor>
          </td>
        </tr>
        <tr>
          <td>Emotion</td>
          <td>
            <Anchor href="https://emotion.sh/docs/composition">Docs</Anchor>
          </td>
          <td>
            <Anchor href="https://github.com/emotion-js/emotion">Github</Anchor>
          </td>
        </tr>
        <tr>
          <td>Create React App</td>
          <td>
            <Anchor href="https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#table-of-contents">
              Docs
            </Anchor>
          </td>
          <td>
            <Anchor href="https://github.com/facebook/create-react-app/tree/next/packages/react-scripts">
              Github
            </Anchor>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
);

export default Home;
