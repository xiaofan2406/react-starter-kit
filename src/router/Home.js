import React from 'react';
import { css } from 'emotion';

const cssClass = css`
  font-size: 48px;
`;

function Home() {
  return <div className={cssClass}>Hello world!</div>;
}

export default Home;
