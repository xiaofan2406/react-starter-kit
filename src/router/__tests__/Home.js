import React from 'react';
import { shallow } from 'enzyme';

import Home from '../Home';

it('should match snapshot', () => {
  const component = shallow(<Home />);

  expect(component).toMatchSnapshot();
});
