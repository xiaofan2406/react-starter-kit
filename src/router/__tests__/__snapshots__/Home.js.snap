// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="css-11o64f7"
>
  Hello world!
</div>
`;
